### PR TITLE
Winter refactoring: Refactor formatting strings.

### PR DESCRIPTION
**Pull request by tfiedor**
This pull request is part of winter refactoring.

1. It transforms most of `.format()` to `f""` formatting string notation;
2. It removes most of the `print` statements in favour of `log.info`;
3. It polishes some minor things.

### CODE REVIEW

File: perun/logic/config.py [hunk 1]
```diff
@@ -173,7 +173,6 @@ def write_config_to(path: str, config_data: dict[str, Any]) -> None:
     :param str path: path where the config will be stored to
     :param dict config_data: dictionary with contents of the configuration
     """
-    perun_log.msg_to_stdout("Writing config '{}' at {}".format(config_data, path), 2)
```
Comment (JiriPavela) [inline perun/logic/config.py:176]: There seems to be no replacement for this message. Is this intended or just a mistake?

File: perun/logic/index.py [hunk 2]
```diff
@@ -448,11 +445,8 @@ def write_entry_to_index(index_file: str, file_entry: BasicIndexEntry) -> None:
                     looked_up_entry.path == file_entry.path
                     and looked_up_entry.time == file_entry.time
                 ):
-                    perun_log.msg_to_stdout(
-                        "{0.path} ({0.time}) already registered in {1}".format(
-                            file_entry, index_file
-                        ),
-                        0,
+                    perun_log.warn(
+                        "{file_entry.path} ({file_entry.time}) already registered in {index_file}",
```
Comment (JiriPavela) [inline perun/logic/index.py]: Seems like this should be a f-string.

File: perun/view/raw/run.py [hunk 3]
```diff
@@ -26,22 +26,19 @@ def show(profile: Profile, **kwargs: Any) -> None:
...
-        print(log.in_color("collector:", RAW_KEY_COLOUR))
+        log.info(log.in_color("collector:", RAW_KEY_COLOUR))
         collector_info = profile["collector_info"]
         for collector_item in ["name", "params"]:
             if collector_item in collector_info.keys():
-                print(
+                log.info(
                     int(raw_indent) * 1 * " "
                     + "- {}: {}".format(
                         log.in_color(collector_item, RAW_KEY_COLOUR),
```
Comment (JiriPavela) [inline perun/view/raw/run.py:44]: Seems like one `.format` string got through here.

**Review by JiriPavela — APPROVED** [review]
Good job, the codebase seems to be a bit more readable (and should be faster a bit as well) now when f-strings are used instead of `format`. Please look into the minor remarks I added in the review before merging.

File: perun/logic/config.py (hunk 1 above)
Comment (tfiedor) [reply perun/logic/config.py:176]: Yes, it was intended.

File: perun/view/raw/run.py (hunk 3 above)
Comment (tfiedor) [reply perun/view/raw/run.py:44]: Intended.

**Pull request merged**